host/3.0/buster/amd64/node/node14/node14-slim.Dockerfile

```diff
@@ -1,33 +1,5 @@
-# Build the runtime from source
 ARG HOST_VERSION=3.0.15584
-FROM mcr.microsoft.com/dotnet/core/sdk:3.1 AS runtime-image
-ARG HOST_VERSION
-
-ENV PublishWithAspNetCoreTargetManifest=false
-
-RUN BUILD_NUMBER=$(echo ${HOST_VERSION} | cut -d'.' -f 3) && \
-    git clone --branch v${HOST_VERSION} https://github.com/Azure/azure-functions-host /src/azure-functions-host && \
-    cd /src/azure-functions-host && \
-    HOST_COMMIT=$(git rev-list -1 HEAD) && \
-    dotnet publish -v q /p:BuildNumber=$BUILD_NUMBER /p:CommitHash=$HOST_COMMIT src/WebJobs.Script.WebHost/WebJobs.Script.WebHost.csproj -c Release --output /azure-functions-host --runtime linux-x64 && \
-    mv /azure-functions-host/workers /workers && mkdir /azure-functions-host/workers && \
-    rm -rf /root/.local /root/.nuget /src
-
-RUN EXTENSION_BUNDLE_VERSION=1.6.0 && \
-    EXTENSION_BUNDLE_FILENAME=Microsoft.Azure.Functions.ExtensionBundle.1.6.0_linux-x64.zip && \
-    apt-get update && \
-    apt-get install -y gnupg wget unzip && \
-    wget https://functionscdn.azureedge.net/public/ExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/$EXTENSION_BUNDLE_VERSION/$EXTENSION_BUNDLE_FILENAME && \
-    mkdir -p /FuncExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/$EXTENSION_BUNDLE_VERSION && \
-    unzip /$EXTENSION_BUNDLE_FILENAME -d /FuncExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/$EXTENSION_BUNDLE_VERSION && \
-    rm -f /$EXTENSION_BUNDLE_FILENAME && \
-    EXTENSION_BUNDLE_VERSION_V2=2.4.0 && \
-    EXTENSION_BUNDLE_FILENAME_V2=Microsoft.Azure.Functions.ExtensionBundle.${EXTENSION_BUNDLE_VERSION_V2}_linux-x64.zip && \
-    wget https://functionscdn.azureedge.net/public/ExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/$EXTENSION_BUNDLE_VERSION_V2/$EXTENSION_BUNDLE_FILENAME_V2 && \
-    mkdir -p /FuncExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/$EXTENSION_BUNDLE_VERSION_V2 && \
-    unzip /$EXTENSION_BUNDLE_FILENAME_V2 -d /FuncExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/$EXTENSION_BUNDLE_VERSION_V2 && \
-    rm -f /$EXTENSION_BUNDLE_FILENAME_V2 &&\
-    find /FuncExtensionBundles/ -type f -exec chmod 644 {} \;
+FROM functionshost:${HOST_VERSION} AS runtime-image
```

Comment:
Is it possible to pass the version of node to install as a parameter to the base image so that node installation can also be moved up?

Reply:
I chose to hold off of making changes like this because the base image in this commit is intended for reuse in all other languages.  But, I am open to discussion about changing that. I just felt that it would be cleaner to have a single base image that is common to all languages. 

It is definitely possible to pass in node versions/any other versioning as an ARG during build.  I am also looking into Tsuyoshi/Bala's suggestion of using a tool to template the images at build time rather than maintaining all of the dockerfiles.